### PR TITLE
fix: load empty bucket's bug v0.2.3

### PR DESCRIPTION
**Pull request by jingjunLi**
### Description

When restarting the service and loading the database, in the case of an empty bucket, it is necessary to send a transaction to the blockchain.

### Rationale

NA

### Example

NA

### Changes

Notable changes: 
* NA
* ...

### CODE REVIEW

**Review by alokeutpal — APPROVED** [review]

File: modular/manager/bucket_migrate_scheduler.go [hunk 1]
```diff
@@ -729,52 +761,51 @@ func (s *BucketMigrateScheduler) loadBucketMigrateExecutePlansFromDB() error {
...
+		if _, found := migrationBucketEventsMap[bucketID]; !found {
+			return fmt.Errorf("migrate bucket events is not found, bucketID:%d", bucketID)
 		}
-		migrateGVGUnitMeta, err = s.manager.baseApp.GfSpDB().ListMigrateGVGUnitsByBucketID(bucketID)
+		bucketMigrateEvent := migrationBucketEventsMap[bucketID]
+
+		primarySPGVGList, err = s.manager.baseApp.GfSpClient().ListGlobalVirtualGroupsByBucket(context.Background(), bucketID)
 		if err != nil {
-			return err
+			log.Errorw("failed to list gvg", "error", err)
```
Comment (will-2012) [inline modular/manager/bucket_migrate_scheduler.go:782]: Add more context, such as bucket_id?

File: modular/manager/bucket_migrate_scheduler.go [hunk 2]
```diff
@@ -729,52 +761,51 @@ func (s *BucketMigrateScheduler) loadBucketMigrateExecutePlansFromDB() error {
...
 
 		log.Debugw("bucket migrate scheduler load from db", "executePlan", executePlan, "bucketID", bucketID)
+		if err = executePlan.Start(); err != nil {
+			log.Errorw("failed to start bucket migrate execute plan", "Events", bucketMigrateEvent.Events, "executePlan", executePlan, "error", err)
+			return err
+		}
 		s.executePlanIDMap[executePlan.bucketID] = executePlan
 	}
-	log.Debugw("Bucket Migrate Scheduler load from db success", "bucketIDs", bucketIDs)
+	log.Debugw("Bucket Migrate Scheduler load from db success", "bucketIDs", migratingBucketIDs)
```
Comment (sysvm) [inline modular/manager/bucket_migrate_scheduler.go:808]: use upper case to print logs.

File: modular/manager/bucket_migrate_scheduler.go [hunk 3]
```diff
@@ -729,52 +761,51 @@ func (s *BucketMigrateScheduler) loadBucketMigrateExecutePlansFromDB() error {
...
+		} else {
+			executePlan, err = s.generateBucketMigrateGVGExecuteUnitFromDB(primarySPGVGList, bucketID, executePlan)
+			if err != nil {
+				log.Errorw("failed to produce bucket migrate execute plan", "Events", bucketMigrateEvent.Events, "error", err)
+				return err
 			}
 		}
 
 		log.Debugw("bucket migrate scheduler load from db", "executePlan", executePlan, "bucketID", bucketID)
+		if err = executePlan.Start(); err != nil {
```
Comment (will-2012) [inline modular/manager/bucket_migrate_scheduler.go:802]: What happens to an empty bucket plan start?

File: modular/manager/bucket_migrate_scheduler.go [hunk 4]
```diff
@@ -533,6 +533,38 @@ func (s *BucketMigrateScheduler) replaceExitingSP(secondarySPIDs []uint32) ([]ui
...
+	migrateGVGUnitMeta, err := s.manager.baseApp.GfSpDB().ListMigrateGVGUnitsByBucketID(bucketID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Using migrateGVGUnitMeta to construct PrimaryGVGIDMapMigrateUnits and execute them one by one.
+	for _, migrateGVG := range migrateGVGUnitMeta {
+		srcSP, queryErr := s.manager.virtualGroupManager.QuerySPByID(migrateGVG.SrcSPID)
+		if queryErr != nil {
+			log.Errorw("failed to query sp", "error", queryErr)
```
Comment (will-2012) [inline modular/manager/bucket_migrate_scheduler.go:550]: Add sp_id context.

File: modular/manager/bucket_migrate_scheduler.go [hunk 5]
```diff
@@ -533,6 +533,38 @@ func (s *BucketMigrateScheduler) replaceExitingSP(secondarySPIDs []uint32) ([]ui
...
+			return nil, queryErr
+		}
+		destSP, queryErr := s.manager.virtualGroupManager.QuerySPByID(migrateGVG.DestSPID)
+		if queryErr != nil {
+			log.Errorw("failed to query sp", "error", queryErr)
```
Comment (will-2012) [inline modular/manager/bucket_migrate_scheduler.go:555]: Ditto.

**Review by will-2012 — APPROVED** [review]

**Review by Fynnss — APPROVED** [review]

**Pull request merged**